### ISSUE TITLE
prevent orphaned dosobjs

### ISSUE THREAD

**Issue by stsp** [stsp/fdpp#91]
As was mentioned in https://github.com/stsp/fdpp/issues/85#issuecomment-512910652
things like `MK_FAR_STR_OBJ()` may not tidy up properly,
for example if the parent object is never actually created and
the child is left orphaned.
I don't think there is currently the code that can lead to that
condition, but such condition is not checked, so the concern
is valid.

Possible solutions:
1. Rearrange the user code (in freedos) the way that the
child object's lifetime overlaps with the parent's. Then the
child object can assert if the parent didn't pick it.
2. In `objtrace_gc()` just assert that the pending child lists
are empty. If any false-positives appear (very unlikely), fix
them by hands.

2 looks simple.

**Comment by stsp** [comment]
Looks very difficult.
1 doesn't work because if you create the child
shared, as is now, its lifetime is not tracked. If
create it non-shared, then it won't be possible
to transfer ownership to parent. And this is only
in a simple case. In non-simple case you'll also
have to rearrange the freedos code before facing
that problem.
2 doesn't work in general because the children
may be pending for very long (like cfgInitTail of
Config), but may work in some limited cases.
Which will mean yet another tracker list for those
limited cases...
Basically doesn't seem to be worth an efforts.
